GitUI/Editor/FileViewer.cs

```diff
@@ -982,10 +988,12 @@ private void ResetView(ViewMode viewMode, string? fileName, FileStatusItem? item
                 _viewMode = ViewMode.FixedDiff;
             }
 
-            SupportLinePatching = ((IsDiffView(_viewMode) && (text?.Contains("@@") ?? false))
-                        || (item is not null && item.Item.IsNew && (item.Item.Staged is StagedStatus.WorkTree or StagedStatus.Index)))
-                    && !Module.IsBareRepository()
-                    && File.Exists(_fullPathResolver.Resolve(fileName));
+            SupportLinePatching = ((IsDiffView(_viewMode) && (text?.Contains("@@") ?? false)
+                        && File.Exists(_fullPathResolver.Resolve(fileName)))
+                    || ((item?.Item.IsNew ?? false)
+                        && ((item.Item.Staged is StagedStatus.WorkTree or StagedStatus.Index)
+                            || !File.Exists(_fullPathResolver.Resolve(fileName)))))
+                && !Module.IsBareRepository();
```

Comment:
This doesn't look readable at all. Can we please break it up and use temp variables?
```cs
if (!Module.IsBareRepository())
{
    SupportLinePatching = false;
}
else 
{
    bool ... = IsDiffView(_viewMode) && (text?.Contains("@@") ?? false) && File.Exists(_fullPathResolver.Resolve(fileName));
    bool ... = (item?.Item.IsNew ?? false) && ((item.Item.Staged is StagedStatus.WorkTree or StagedStatus.Index) || !File.Exists(_fullPathResolver.Resolve(fileName))));
    SupportLinePatching = ... || ...;
}
```

Reply:
Some comments could help too

Reply:
Added comments
I prefer one assignment, to make sure the more expensive calls to File.Exists() is just executed once.